resteasy-core/src/main/java/org/jboss/resteasy/core/providerfactory/ResteasyProviderFactoryImpl.java

```diff
@@ -1476,7 +1462,7 @@ public T getContext(Class type)
    @Override
    public Map<String, Object> getProperties()
    {
-      return Collections.unmodifiableMap(properties);
+      return properties;

```

Comment:
Is this really needed? I mean I believe the idea here was that getProperties returns an unmodifiable map, while getMutableProperies returns the actual mutable map. Is there a problem with this?

Reply:
Thanks, I'll be merging soon!